File: Daily_knowledge_set/Daily_knowledge_set/Classes/SwiftString/SwiftStringViewController.m
//
//  OCStringViewController.m
//  Daily_knowledge_set
//
//  Created by Jason on 25/01/2017.
//  Copyright © 2017 Jason. All rights reserved.
//

#import "SwiftStringViewController.h"

@interface SwiftStringViewController ()

@end

@implementation SwiftStringViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    [self forEachInObjc];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)forEachInObjc {
    NSArray *datas = @[@1,@2,@3];
    
    for (int i = 0; i < datas.count; ++i) {
        NSLog(@"forI:%@",datas[i]);
    }
    
    for (NSNumber *d in datas) {
        NSLog(@"forin:%@",d);
    }
    
    [datas enumerateObjectsUsingBlock:^(id  _Nonnull d, NSUInteger idx, BOOL * _Nonnull stop) {
        NSLog(@"enumerate:%@",d);
    }];
}

- (void)forEachStringInObjc {
    NSString *s = @"testString";
    
    // forI
    for (int i = 0; i < s.length; ++i) {
        NSLog(@"%c",[s characterAtIndex:i]);
    }
}





@end

File: Daily_knowledge_set/Daily_knowledge_set/Classes/SwiftString/SwiftStringViewController.swift
//
//  SwiftViewController.swift
//  Daily_knowledge_set
//
//  Created by Jason on 28/01/2017.
//  Copyright © 2017 Jason. All rights reserved.
//

import UIKit

class SwiftStringViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = #colorLiteral(red: 1, green: 1, blue: 1, alpha: 1)
        
        var s = "Swift"
        print(s)
        
        s.replaceSubrange(s.index(s.startIndex,offsetBy:1)...s.index(s.startIndex, offsetBy: 4), with: "WIFT")
        
        s = "Swift"
        print(s)
        print("forI")
        for i in 0..<s.characters.count {
            print(s.characters[s.characters.index(s.startIndex, offsetBy: i)])
        }
        print("forin")
        for c in s.characters {
            print(c)
        }
        print("block")
        s.characters.forEach { (c) in
            print(c)
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

}
